vmm/src/device_manager/mmio.rs

```diff
@@ -341,9 +350,11 @@ mod tests {
         // drive. The purpose of this method is to test error scenarios and should otherwise
         // not be used.
         pub fn remove_device_info(&mut self, type_id: u32, id: &str) {
-            self.id_to_dev_info
-                .remove(&(DeviceType::Virtio(type_id), id.to_string()))
-                .unwrap();
+            let index = self.id_to_dev_info
+                            .iter()
+                            .position(|x| x.dev_type == DeviceType::Virtio(type_id) && x.dev_id == id.to_string())
```

Comment:
```suggestion
                            .position(|x| x.dev_type == DeviceType::Virtio(type_id) && x.dev_id == id)
```
in order to get rid of the clippy error.